jsf/subsystem/src/main/java/org/jboss/as/jsf/deployment/JSFMetadataProcessor.java

```diff
@@ -34,10 +46,28 @@ public void deploy(DeploymentPhaseContext phaseContext) throws DeploymentUnitPro
                 jsf.setMultipartConfig(new MultipartConfigMetaData());
             }
         }
-
+        if (disallowDoctypeDecl != null) {
+            // Add the disallowDoctypeDecl context param if it's not already present
+            setContextParameterIfAbsent(metaData.getMergedJBossWebMetaData(), DISALLOW_DOCTYPE_DECL, disallowDoctypeDecl.toString());
+        }
     }
 
     @Override
     public void undeploy(DeploymentUnit context) {
     }
+
+    private void setContextParameterIfAbsent(final JBossWebMetaData webMetaData, final String name, final String value) {
+        List<ParamValueMetaData> contextParams = webMetaData.getContextParams();
+        if (contextParams == null) {
+            contextParams = new ArrayList<ParamValueMetaData>();
+            webMetaData.setContextParams(contextParams);
+        }
+        boolean paramFound = contextParams.stream().anyMatch(param -> name.equals(param.getParamName()) && param.getParamValue() != null);
```

Comment:
Please switch to a basic for loop. Lambdas are expensive in terms of class storage space.